crates/red_knot_python_semantic/src/types.rs

```diff
@@ -424,27 +416,35 @@ impl<'db> Type<'db> {
             (Type::Never, _) => true,
             (_, Type::Never) => false,
             (Type::IntLiteral(_), Type::Instance(class))
-                if class.is_stdlib_symbol(db, "builtins", "int") =>
+                if matches!(class.is_builtin(db), Some(BuiltinType::Int)) =>
```

Comment:
Can we add a method on `ClassType` to make this less verbose? I would probably even call it `is_builtin`:
```suggestion
                if class.is_builtin(db, BuiltinType::Int) =>
```

And then rename the `is_builtin` field to just `builtin`; `is_` prefix suggests a boolean, not an optional enum.